### PR TITLE
[AWS] Enable p4de in the catalog

### LINKED ISSUE #1201: p4de.24xlarge not supported
Hi, I'm trying to run an experiment using a `p4de.24xlarge` instance on AWS, but I'm unable to launch the instance using skypilot. I have access to the instance via AWS on us-west-2, and am able to launch it via the web interface.

I tried the following commands and ran into the following errors:
```
$ sky gpunode --cloud aws --instance-type p4de.24xlarge
ValueError: Invalid instance type 'p4de.24xlarge' for cloud AWS.
```

```
$ sky launch --instance-type p4de.24xlarge -c benchmark-a100 skypilot_configs/a100.yaml
W 10-05 17:23:38 resources.py:573] image_id in resources is experimental. It only supports AWS/GCP.
ValueError: Invalid instance type 'p4de.24xlarge' for cloud AWS
```

`p4de.24xlarge` also doesn't appear when searching `sky show-gpus --all`.

More info on the p4de.24xlarge:
https://aws.amazon.com/about-aws/whats-new/2022/05/amazon-ec2-p4de-gpu-instances-ml-training-hpc/

### PR DESCRIPTION
**Pull request by Michaelvll**
Since our AWS credential does not have permission to query the offerings for A100-80GB instances, we now hardcoded it in our `fetch_aws` to unblock the uses of that instance for users.

Fixes #1201

Tested (run the relevant ones):

- [x] Any manual or new tests for this PR (please specify below)
  - [x] `python -m sky.clouds.service_catalog.data_fetchers.fetch_aws`
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py [hunk 1]
```diff
@@ -166,6 +168,34 @@ def _get_spot_pricing_table(region: str) -> pd.DataFrame:
...
+        df = df.append(pd.Series({
+            'InstanceType': 'p4de.24xlarge',
+            'AcceleratorName': 'A100-80GB',
+            'AcceleratorCount': 8,
+            'vCPUs': 96,
+            'MemoryGiB': 1152,
+            'GpuInfo':
+                ('{\'Gpus\': [{\'Name\': \'A100-80GB\', \'Manufacturer\': '
+                 '\'NVIDIA\', \'Count\': 8, \'MemoryInfo\': {\'SizeInMiB\': '
+                 '40960}}], \'TotalGpuMemoryInMiB\': 327680}'),
```
Comment (concretevitamin) [inline sky/clouds/service_catalog/data_fetchers/fetch_aws.py:188]: Should these memory be ~80GB?

Do we have documentation somewhere on the diff b/t SizeInMiB and TotalGpuMemoryInMiB?

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py [hunk 2]
```diff
@@ -59,6 +60,7 @@
 # only available in this region, but it serves pricing information for all
 # regions.
 PRICING_TABLE_URL_FMT = 'https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/{region}/index.csv'  # pylint: disable=line-too-long
+P4DE_REGIONS = ['us-east-1', 'us-west-2']
```
Comment (concretevitamin) [inline sky/clouds/service_catalog/data_fetchers/fetch_aws.py:63]: Doc, add ref https://aws.amazon.com/ec2/instance-types/p4/?

**Review by concretevitamin — COMMENTED** [review]
Thanks @Michaelvll, some questions.

Also: after users upgrade past this PR, do they need to do anything, or do they have to wait till up to 7 hr for the next catalog refresh? I think it's ok, just checking.

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py (hunk 2 above)
Comment (Michaelvll) [reply sky/clouds/service_catalog/data_fetchers/fetch_aws.py:66]: Done. Thanks!

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py (hunk 1 above)
Comment (Michaelvll) [reply sky/clouds/service_catalog/data_fetchers/fetch_aws.py]: Ahh good catch! Fixed. Thanks!

The `SizeInMiB` seems to be the memory of a single GPU and `TotalGpuMemoryInMiB` is the total memory for 8 GPUs

**Comment by concretevitamin** [comment]
> Also: after users upgrade past this PR, do they need to do anything, or do they have to wait till up to 7 hr for the next catalog refresh? I think it's ok, just checking.

One problem: I ran the fetcher locally, but the fetched catalog doesn't contain any A100-80GB.

**Comment by Michaelvll** [comment]
> Also: after users upgrade past this PR, do they need to do anything, or do they have to wait till up to 7 hr for the next catalog refresh? I think it's ok, just checking.

Oops, missed this. Yes, the user does not have to do anything. It will be updated automatically.

> One problem: I ran the fetcher locally, but the fetched catalog doesn't contain any A100-80GB.

That is weird, it works with my account. Could you try the same command as we run in the catalog repo `python -m sky.clouds.service_catalog.data_fetchers.fetch_aws --no-az-mappings --check-all-regions-enabled-for-account`?

**Comment by concretevitamin** [comment]
```
python -m sky.clouds.service_catalog.data_fetchers.fetch_aws
```
succeeded but did not produce A100-80GB.

```
 python -m sky.clouds.service_catalog.data_fetchers.fetch_aws --no-az-mappings --check-all-regions-enabled-for-account
```
failed with
```
RuntimeError: The following regions are not enabled: {'eu-south-1', 'af-south-1', 'me-central-1', 'ap-southeast-3', 'me-south-1', 'ap-east-1'}
```

Approving for now to unblock potential usage.

**Review by concretevitamin — APPROVED** [review]

**Comment by Michaelvll** [comment]
Merging this for now to unblock the users trying AWS for the A100-80GB instances. Let us debug offline @concretevitamin. : )

**Comment by Michaelvll** [comment]
Just confirmed in the catalog repo, the `p4de.24xlarge` is included in the catalog file now.

**Pull request merged**